toolkit/tools/internal/network/network.go

```diff
@@ -62,12 +146,12 @@ func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certif
 
 	response, err := client.Get(url)
 	if err != nil {
-		return
+		return fmt.Errorf("%w:\nrequest failed:\n%w", ErrDownloadFileOther, err)
```

Comment:
While here, we should also make use of `http.NewRequestWithContext`. On way to do it, is to replace `client.Get(url)` with:

```golang
	request, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
	if err != nil {
		return fmt.Errorf("%w:\nfailed to create request:\n%w", ErrDownloadFileOther, err)
	}

	response, err := client.Do(request)
```